app/models/course_user.rb

```diff
@@ -36,7 +36,12 @@ class CourseUser < ActiveRecord::Base
                                        inverse_of: :course_user, dependent: :destroy
   has_many :course_user_achievements, class_name: Course::UserAchievement.name,
                                       inverse_of: :course_user, dependent: :destroy
-  has_many :achievements, through: :course_user_achievements, class_name: Course::Achievement.name
+  has_many :achievements, through: :course_user_achievements,
+                          class_name: Course::Achievement.name do
+    def ordered_by_date_obtained
+      order('course_user_achievements.obtained_at desc')
```

Comment:
Sorry actually this is not a calculated attribute >_< might have responded wrongly because I was trying to do it on mobile. But this is applicable for the comment below. 

I tried the squeel suggestion and it works so I will do that.